From 51d9936baf27573e45dcfa68e2bcb3624e6c1f50 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Thu, 31 Mar 2011 14:44:50 +0100
Subject: [PATCH] Use the system version of rply if available

Fedora package review requires that packages do not bundle other libraries, and
instead use the system copies where available.

This patch allows mash to use the system version of rply if it is installed,
otherwise the internal copy is used like before.
---
 configure.ac     |    8 ++++++++
 mash/Makefile.am |   12 +++++++++++-
 2 files changed, 19 insertions(+), 1 deletions(-)

diff -urNp mash-0.1.0.old/configure.ac mash-0.1.0/configure.ac
--- mash-0.1.0.old/configure.ac	2011-03-31 14:54:47.923819594 +0100
+++ mash-0.1.0/configure.ac	2011-03-31 15:05:06.965819437 +0100
@@ -78,6 +78,20 @@ AS_IF([test "x$have_mx" = "xno"],
 
 AM_CONDITIONAL([HAVE_MX], [test "x$have_mx" = "xyes"])
 
+dnl Prefer the system version of rply if available
+AC_CHECK_HEADER([rply/rply.h], [have_rply=yes], [have_rply=no])
+
+if test x$have_rply = "xno"; then
+        AC_MSG_NOTICE([rply was not found. Using internal copy])
+else
+        RPLY_CFLAGS=""
+        RPLY_LIBS="-lrply"
+        AC_SUBST(RPLY_CFLAGS)
+        AC_SUBST(RPLY_LIBS)
+fi
+
+AM_CONDITIONAL([USE_INTERNAL_RPLY], [test "x$have_rply" = "xno"])
+
 # prefixes for fixing gtk-doc references
 CLUTTER_PREFIX="`$PKG_CONFIG --variable=prefix clutter-1.0`"
 AC_SUBST(CLUTTER_PREFIX)
diff -urNp mash-0.1.0.old/mash/Makefile.am mash-0.1.0/mash/Makefile.am
--- mash-0.1.0.old/mash/Makefile.am	2011-03-31 14:54:47.922819594 +0100
+++ mash-0.1.0/mash/Makefile.am	2011-03-31 15:05:06.965819437 +0100
@@ -1,4 +1,6 @@
+if USE_INTERNAL_RPLY
 SUBDIRS = rply
+endif
 
 lib_LTLIBRARIES = libmash-@MASH_API_VERSION@.la
 
@@ -7,6 +9,7 @@ INCLUDES = \
 
 AM_CPPFLAGS = \
 	-DMASH_COMPILATION=1 \
+	@RPLY_CFLAGS@ \
 	@CLUTTER_CFLAGS@
 
 enum_h = $(srcdir)/mash-data.h
@@ -53,8 +56,15 @@ libmash_@MASH_API_VERSION@_la_LDFLAGS = 
 	-version-info "@MASH_LT_CURRENT@:@MASH_LT_REVISION@:@MASH_LT_AGE@"
 
 libmash_@MASH_API_VERSION@_la_LIBADD = \
-	@CLUTTER_LIBS@ \
+	@CLUTTER_LIBS@
+
+if USE_INTERNAL_RPLY
+libmash_@MASH_API_VERSION@_la_LIBADD += \
 	rply/librply.la
+else
+libmash_@MASH_API_VERSION@_la_LIBADD += \
+	@RPLY_LIBS@
+endif
 
 CLEANFILES =
 
-- 
1.7.4.2

